mac/hidapi_darwin.h

```diff
@@ -0,0 +1,49 @@
+/*******************************************************
+ HIDAPI - Multi-Platform library for
+ communication with HID devices.
+
+ libusb/hidapi Team
+
+ Copyright 2022, All Rights Reserved.
+
+ At the discretion of the user of this library,
+ this software may be licensed under the terms of the
+ GNU General Public License v3, a BSD-Style license, or the
+ original HIDAPI license as outlined in the LICENSE.txt,
+ LICENSE-gpl3.txt, LICENSE-bsd.txt, and LICENSE-orig.txt
+ files located at the root of the source distribution.
+ These files may also be found in the public source
+ code repository located at:
+        https://github.com/libusb/hidapi .
+********************************************************/
+
+/** @file
+ * @defgroup API hidapi API
+ */
+
+#ifndef HIDAPI_DARWIN_H__
+#define HIDAPI_DARWIN_H__
+
+#include <stdint.h>
+
+#include "hidapi.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+		/** @brief Get the location ID for a HID device.
+
+			@ingroup API
+			@param path The path name of the device
+
+			@returns
+				This function returns the location ID, or 0 on error.
+		*/
+		unsigned HID_API_EXPORT_CALL hid_darwin_get_location_id(const char *path);
```

Comment:
make it a:

`int HID_API_EXPORT_CALL hid_darwin_get_location_id(hid_device *dev, uint32_t *location_id);`

returns `-1` on failure - like all other function

---
and if someone wants to use it with `path`: use `hid_open_path` -> `hid_darwin_get_location_id` -> `hid_close`